### PR TITLE
 brasero: rename brasero-original to brasero-unwrapped; rhythmbox: use brasero-unwrapped instead of brasero

### PR DESCRIPTION
**Pull request by fabianhjr**
## Description of changes

Also mark brasero-wrapped as unfree due to cdrtools license incompat with GPL

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by flokli** [comment]
Is libburn a library essentially accomplishing the same, but with less license uncertainty? Or is some functionality limited to libdvdcss? Maybe there's some explanation upstream we can link to? 

In case it doesn't fully replace functionality, I'd propose adding the libdvdcss LD_PRELOAD in a fixupPhase, configurable by a enableLibdvdcss, which we can disable by default.

**Comment by fabianhjr** [comment]
afaik libburn is one of the three supported backbends by brasero according to their site

https://wiki.gnome.org/Apps/Brasero

Unsure if as feature complete

**Comment by flokli** [comment]
libdvdcss is also used by libdvdread (both hosted by the VideoLAN project these days). It is widely used in vlc, mpv and various other players and tools.

> Drop cdrtools since it is non-redistributable due to license issues and brasero is used as a lib for example by rythmbox.

libdvdcss is gpl2, which is redistributable. I don't think you're referring to license issues, but legal issues.
https://www.videolan.org/legal.html has a section about libdvdcss and libaacs.

nixpkgs provides recipes for building vlc, mpv and everything underneath, but it's up to the user to configure their system in a way that adds such components.

**Comment by fabianhjr** [comment]
https://en.wikipedia.org/wiki/Cdrtools#License_compatibility_controversy

https://en.wikipedia.org/wiki/Common_Development_and_Distribution_License

^ these are regarding cdrtools license issues

**Review by flokli — CHANGES_REQUESTED** [review]
I'm okay with dropping cdrtools if brasero can use another backend, but I don't think

> Downstream users are able to wrap brasero as providad by nixpkgs with both of these dependencies

is okay for simply removing libdvdcss support.

A lot of DVDs out there use it CSS, it's been known to work with brasero for a long time, and having this suddenly regress would be quite annoying.

I researched a bit, brasero support in rhythmbox seems to be only used to burn optical media, not read from it, and CSS isn't applicable to Audio CDs anyways.

So what do you think about replacing `brasero` with `brasero-original` in the `rhythmbox` derivation (which should really be called `brasero-unwrapped`)?

That should also prevent you from pulling in libdvdcss, without regressing it for people using brasero.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -4167,9 +4167,9 @@ with pkgs;
 
   bozohttpd-minimal = bozohttpd.override { minimal = true; };
 
-  brasero-original = lowPrio (callPackage ../tools/cd-dvd/brasero { });
+  brasero = callPackage ../tools/cd-dvd/brasero { };
```
Comment (flokli) [inline pkgs/top-level/all-packages.nix:4170]: This would be brasero-unwrapped.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -4167,9 +4167,9 @@ with pkgs;
...
 
-  brasero = callPackage ../tools/cd-dvd/brasero/wrapper.nix { };
+  brasero-wrapped = callPackage ../tools/cd-dvd/brasero/wrapper.nix { };
```
Comment (flokli) [inline pkgs/top-level/all-packages.nix:4172]: .. and this just brasero. Compare with other examples in nixpkgs, like Firefox and chromium.

**Review by flokli — DISMISSED** [review]

File: pkgs/tools/cd-dvd/brasero/wrapper.nix [hunk 3]
```diff
@@ -14,5 +14,8 @@ in symlinkJoin {
       --prefix LD_PRELOAD : ${lib.makeLibraryPath [ libdvdcss ]}/libdvdcss.so
   '';
 
-  inherit (brasero-original) meta;
+  meta = brasero-unwrapped.meta // {
+    # - cdrtools is non-redistributable when mixed with GPL code
```
Comment (flokli) [inline pkgs/tools/cd-dvd/brasero/wrapper.nix:18]: Can we remove cdrtools from the wrapper entirely, assuming the libburn backend is sufficient (this is a question, not a request for change)?

Independent of that, I'd get why linking might be problematic, but this simply adds cdrtools to $PATH, it doesn't link, neither static nor dynamically.

File: pkgs/applications/audio/rhythmbox/default.nix [hunk 4]
```diff
@@ -21,7 +21,7 @@
 , libgpod
 , libmtp
 , lirc
-, brasero
+, brasero-unwrapped
```
Comment (flokli) [inline pkgs/applications/audio/rhythmbox/default.nix:24]: Add a comment you're using brasero-unwrapped so you don't pull in libdvdcss which isn't used by rhythmbox.

File: pkgs/tools/cd-dvd/brasero/wrapper.nix (hunk 3 above)
Comment (emilazy) [reply pkgs/tools/cd-dvd/brasero/wrapper.nix:18]: The comment here is inaccurate. The problem is that `cdrtools` is itself a combination of GPL and CDDL code.

I agree that we should keep `libdvdcss` and just drop `cdrtools` if possible (most distributions don’t even package `cdrtools` and we probably shouldn’t bother either).

Comment (flokli) [reply pkgs/tools/cd-dvd/brasero/wrapper.nix:18]: Yeah, in that case cdrtools should just be marked as non-redistributable, unfree or whatever, and it'd propagate into here. But as I said, if brasero works fine without cdrtools, we can just drop it from here.

Comment (fabianhjr) [reply pkgs/tools/cd-dvd/brasero/wrapper.nix]: Dropped this commit

Comment (emilazy) [reply pkgs/tools/cd-dvd/brasero/wrapper.nix]: Looks like `cdrtools` is still there, though?

Comment (flokli) [reply pkgs/tools/cd-dvd/brasero/wrapper.nix]: Yeah, guess that's up for a future PR.

Comment (fabianhjr) [reply pkgs/tools/cd-dvd/brasero/wrapper.nix]: Yeah

**Pull request merged**